awswrangler/s3/_write_parquet.py

```diff
@@ -177,18 +181,26 @@ def _to_parquet_distributed(  # pylint: disable=unused-argument
     bucketing: bool = False,
 ) -> List[str]:
     if bucketing:
-        # Add bucket id
+        # Add bucket id to the prefix
         filename_prefix = f"{filename_prefix}_bucket-{df.name:05d}"
     # Create Ray Dataset
     ds = from_modin(df) if isinstance(df, ModinDataFrame) else from_pandas(df)
+    # Repartition into a single block if or writing into a single key or if bucketing is enabled
+    if ds.count() > 0 and (path or bucketing):
```

Comment:
Is there a case where df.count() <= 0?

Reply:
Yes sometimes happens when partitioning and buketing and there's an empty bucket. Standard implementation writes an empty file in that case. Had to add this check because repartitioning an empty dataset fails